File: 167. 两数之和 II - 输入有序数组/167. 两数之和 II - 输入有序数组.py
class Solution:
    def twoSum(self, nums, target: int):
        # 双指针
        i,j = 0,len(nums)-1
        while i<j:
            if nums[i] + nums[j] == target:
                return [i+1,j+1]
            elif nums[i] + nums[j] > target:
                j -= 1
            else:
                i += 1
        

File: 167. 两数之和 II - 输入有序数组/167. 两数之和 II - 输入有序数组.go
package main

import "fmt"

func main() {
    fmt.Println(twoSum([]int{1,2,3,4,5,6},3))
}

func twoSum(numbers []int, target int) []int {
    i,j := 0,len(numbers)-1
    k := 0
    for ;i<j; {
        k = numbers[i] + numbers[j]
        if k==target {
            return []int{i+1,j+1}
        }else if k < target{
            i += 1
        }else{
            j -= 1
        }
    }
    return []int{i,j}
}